src/lib/model/InboundThirdpartyTransactionModel.js

```diff
@@ -0,0 +1,109 @@
+
+'use strict';
+
+const {
+    BackendRequests,
+    HTTPResponseError,
+} = require('@internal/requests');
+
+const {
+    MojaloopRequests,
+    Ilp,
+    Errors,
+} = require('@mojaloop/sdk-standard-components');
+
+const shared = require('@internal/shared');
+
+/**
+ *  Models the operations required for inbound third party transaction requests
+ */
+class InboundThirdpartyTransactionModel {
+    constructor(config) {
+        this._cache = config.cache;
+        this._logger = config.logger;
+        this._dfspId = config.dfspId;
+        this._expirySeconds = config.expirySeconds;
+
+        this._mojaloopRequests = new MojaloopRequests({
+            logger: this._logger,
+            peerEndpoint: config.peerEndpoint,
+            alsEndpoint: config.alsEndpoint,
+            quotesEndpoint: config.quotesEndpoint,
+            transfersEndpoint: config.transfersEndpoint,
+            dfspId: config.dfspId,
+            tls: config.tls,
+            jwsSign: config.jwsSign,
+            jwsSigningKey: config.jwsSigningKey,
+            wso2Auth: config.wso2Auth
+        });
+
+        this._backendRequests = new BackendRequests({
+            logger: this._logger,
+            backendEndpoint: config.backendEndpoint,
+            dfspId: config.dfspId
+        });
+
+        this._checkIlp = config.checkIlp;
+
+        this._ilp = new Ilp({
+            secret: config.ilpSecret
+        });
+    }
+
+    /**
+     * Queries the backend API to get the authorization details and makes a 
+     * callback to the originator with the result
+     */
+    async postAuthorizations(authorizationsReq, sourceFspId) {
+        try {
+            //Converts a mojaloop authorizationsReq data to internal form
+            const internalForm = shared.mojaloopAuthorizationsReqToInternal(authorizationsReq);
+
+            // make a call to the backend to ask for a authorizations response
+            const response = await this._backendRequests.getSignedChallenge(internalForm);
+            if (!response) {
+                // make an error callback to the source fsp
+                return 'No response from backend';
+            }
+            // project our internal authorizations reponse into mojaloop response form
+            const mojaloopResponse = shared.internalAuthorizationsResponseToMojaloop(response);
+            // make a callback to the source fsp with the party info
+            return this._mojaloopRequests.putAuthorizations(authorizationsReq.transactionRequestId, mojaloopResponse, sourceFspId);
+        }
+        catch (err) {
+            this._logger.push({ err }).log('Error in postAuthorizations');
+            const mojaloopError = await this._handleError(err);
+            this._logger.push({ mojaloopError }).log(`Sending error response to ${sourceFspId}`);
+            return await this._mojaloopRequests.putAuthorizationsError(authorizationsReq.transactionRequestId, mojaloopError, sourceFspId);
```

Comment:
when you are returning from async function there is no need to use the `await` keyword

Reply:
Followed existing pattern.

Reply:
yeah, that is strange, but let keep the existing convention